roxie/ccd/ccdstate.cpp

```diff
@@ -2896,7 +2896,21 @@ extern void loadPlugins()
     if (pluginDirectory.length())
     {
         plugins = new SafePluginMap(&PluginCtx, traceLevel >= 1);
-        plugins->loadFromList(pluginDirectory);
+        if (topology->hasProp("preload"))
+        {
+            Owned<IPropertyTreeIterator> preloads = topology->getElements("preload");
+            ForEach(*preloads)
+            {
+                const char *preload = preloads->query().queryProp(".");
+                if (!streq(preload, "none"))
+                {
+                    VStringBuffer soname(SharedObjectPrefix "%s" SharedObjectExtension, preload);
```

Comment:
Are the plugins guaranteed to be on the path?  Also, what will happen about attempting to load a plugin that is not on the path if it isn't preloaded? 
Also not sure how getAdditionalPluginsPath() will fit into this.

Reply:
I don't think getAdditionalPluginsPath is used any more since we stopped supporting Python 2. The plugins are successfully loaded because Roxie has RUNPATH set to refer to them, so the pluginsPath variable actually ONLY affects what gets preloaded. Could do with some cleaning up though, and I'll have a think about how best this should be done.